core/modules/parsers/wikiparser/rules/fnprocdef.js

```diff
@@ -43,19 +43,14 @@ exports.parse = function () {
 	this.parser.pos = this.matchRegExp.lastIndex;
 	// Parse the parameters
 	var params = [];
-	if (this.match[3]) {
+	if(this.match[3]) {
 		params = $tw.utils.parseParameterDefinition(this.match[4]);
 	}
 	// Is the remainder of the line blank after the parameter close paren?
 	var reEnd;
-	if (this.match[5]) {
+	if(this.match[5]) {
 		// If so, it is a multiline definition and the end of the body is marked with \end
-		reEnd = new RegExp(
-			"((:?^|\\r?\\n)[^\\S\\n\\r]*\\\\end[^\\S\\n\\r]*(?:" +
-				$tw.utils.escapeRegExp(this.match[2]) +
-				")?(?:$|\\r?\\n))",
-			"mg"
-		);
+		reEnd = new RegExp("((:?^|\\r?\\n)[^\\S\\n\\r]*\\\\end[^\\S\\n\\r]*(?:" + $tw.utils.escapeRegExp(this.match[2]) + ")?\\s*?(?:$|\\r?\\n))","mg");
```

Comment:
THen don't merge this before my https://github.com/TiddlyWiki/TiddlyWiki5/pull/8258 , I'm tired of fixing conflicts.

Reply:
hihi. I did merge master because there where conflicts ;) -- This is a draft anyway.